cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -325,13 +322,16 @@ func del(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 			args.ContainerID, err)
 
 		// When IPAMD is unreachable, try to teardown pod network using previous result. This action prevents rules from leaking while IPAMD is unreachable.
-		// Note that connection error is still returned to kubelet so that delete retries until IPAMD can be notified.
+		// Note that no error is returned to kubelet as there is no guarantee that kubelet will retry delete, and returning an error would prevent container runtime
```

Comment:
> There is no guarantee that kubelet will retry delete

May be I am missing something, Why will not kubelet retry deletes? 

For the referenced issue,  https://github.com/aws/amazon-vpc-cni-k8s/issues/2331 was the case for the customer that the IPAM wasn't reachable, and if IPAMD was running these resource limitation won't be hit?

Reply:
Kubelet will not retry deletes if the resource has already been deleted at the API server, which can happen if a raw delete is issued or the pod had a termination grace period of 0 seconds.

#2331 is neither of those cases, though. It is the case where `aws-node` pod is not running on the node, and pods cannot be deleted on the node to make room for it due to IPAMD not running.